Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.py

```diff
@@ -659,6 +743,81 @@ def ediscovery_source_command_results(raw_case_list: list, source_type, raw_res=
                                   to_hr=created_by_fields_to_hr)
 
 
+def query_set_limit(query: str, limit: int) -> str:
+    """
+    Add limit to given query. If the query has limit, changes it.
+    Args:
+        query: the original query
+        limit: new limit value, if the value is negative return the original query.
+    Returns: query with limit parameters
+    """
+    if limit < 0:
+        return query
+
+    # the query has the structure of "section | section | section ..."
+    query_list = re.split(r'(?<!\|)\|(?!\|)', query)
+
+    # split the query to sections and find limit sections
+    for section in query_list:
+        section_list = section.split()
+        # 'take' and 'limit' are synonyms.
+        if section_list and section_list[0] == 'limit' or section_list[0] == 'take':
```

Comment:
please add a comment which explains the condition's logic
(I mean, the design's decision that if the query already contains a limit, so we don't override it

Reply:
I added in doc string